File: Spring Boot Pr(L)/Backend Spring Boot/src/main/java/com/eVote/dto/CandidateDTO.java
package com.eVote.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;

import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;

import jakarta.persistence.Column;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class CandidateDTO {

	private Integer candidateId;

	@NotBlank(message = "Candidate name cannot be blank")
	@Size(max = 255, message = "Candidate name must not exceed 255 characters")
	private String name;

	@Size(max = 2000, message = "Agenda must not exceed 2000 characters")
	private String agenda;

	@NotNull(message = "Poll ID cannot be null")
	private Integer pollId; // Only poll ID is needed to map a candidate to a poll.
	
	@JsonProperty(access = Access.READ_ONLY)
	private LocalDateTime createdAt = LocalDateTime.now();

	@JsonProperty(access = Access.READ_ONLY)
	private String createdBy;

	@JsonProperty(access = Access.READ_ONLY)
	private LocalDateTime updatedAt = LocalDateTime.now();

	@JsonProperty(access = Access.READ_ONLY)
	private String updatedBy;
}

File: ASP.NET Pr(L)/Backend/Backend/DTO/CandidateDto.cs
using Microsoft.AspNetCore.Http.HttpResults;

namespace Backend.DTOs
{
    public class CandidateDto
    {
        public String agenda { get; set; }
        public String name { get; set; }
        public int pollId { get; set; }
        public int userId { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public CandidateDto(String CreatedBy, String UpdatedBy, String agenda, String name, int pollId)
        {
            this.agenda = agenda;
            this.name = name;
            this.pollId = pollId;
            this.CreatedBy = CreatedBy;
            this.UpdatedBy = UpdatedBy;
        }
    }
}
